### PR TITLE
RavenDB-14784 added the ability to load attachment

### PR DESCRIPTION
**Pull request by ppekrol**
TODO:
- add 'GetContentAsStream' so during Lucene conversion we will be able to write byte[] directly without doing any encoding

### CODE REVIEW

File: src/Raven.Client/Documents/Indexes/IAttachmentObject.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+        public string Hash { get; }
+
+        public string ContentType { get; }
+
+        public long Size { get; }
+
+        public string GetContentAsString();
+
+        public string GetContentAsString(Encoding encoding);
```
Comment (ayende) [inline src/Raven.Client/Documents/Indexes/IAttachmentObject.cs:18]: Should we expose a `Stream` here as well?

**Review by ayende — COMMENTED** [review]
I love that in this model, we don't have to deal with refershing the index on load attachment in a special manner.
A change to the attachment _is_ a change to the document and will updte it.!

File: src/Raven.Client/Documents/Indexes/IAttachmentObject.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Indexes/IAttachmentObject.cs:18]: Yes, wrote that in the comment above.

**Comment by ppekrol** [comment]
Done.

**Pull request merged**